bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BUnionType.java

```diff
@@ -120,7 +120,7 @@ public String toString() {
                     sb.append(PIPE);
                 }
             }
-            cachedToString = sb.toString();
+            cachedToString = sb.append(")").toString();
```

Comment:
This was correct the way it was, `typedesc map<int|string>` 
new format is redundant `typedesc map<(int|string)>`.

Only advance case like `(int|boolean[6][4])[2][3]` needs brackets